tests/core/engine_adapter/test_mssql.py

```diff
@@ -0,0 +1,546 @@
+    adapter = make_mocked_engine_adapter(MsSqlEngineAdapter)
+
+    df = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+    adapter.merge(
+        target_table="target",
+        source_table=df,
+        columns_to_types={
+            "id": exp.DataType.Type.INT,
+            "ts": exp.DataType.Type.TIMESTAMP,
+            "val": exp.DataType.Type.INT,
+        },
+        unique_key=["id"],
+    )
+    adapter.cursor.execute.assert_called_once_with(
+        'MERGE INTO "target" AS "__MERGE_TARGET__" USING (SELECT CAST("id" AS INTEGER) AS "id", CAST("ts" AS DATETIME2) AS "ts", CAST("val" AS INTEGER) AS "val" FROM (VALUES (1, 4), (2, 5), (3, 6)) AS "t"("id", "ts", "val")) AS "__MERGE_SOURCE__" ON "__MERGE_TARGET__"."id" = "__MERGE_SOURCE__"."id" '
+        'WHEN MATCHED THEN UPDATE SET "__MERGE_TARGET__"."id" = "__MERGE_SOURCE__"."id", "__MERGE_TARGET__"."ts" = "__MERGE_SOURCE__"."ts", "__MERGE_TARGET__"."val" = "__MERGE_SOURCE__"."val" '
+        'WHEN NOT MATCHED THEN INSERT ("id", "ts", "val") VALUES ("__MERGE_SOURCE__"."id", "__MERGE_SOURCE__"."ts", "__MERGE_SOURCE__"."val")'
+    )
+
+    adapter.cursor.reset_mock()
+    adapter.merge(
+        target_table="target",
+        source_table=df,
+        columns_to_types={
+            "id": exp.DataType.Type.INT,
+            "ts": exp.DataType.Type.TIMESTAMP,
+            "val": exp.DataType.Type.INT,
+        },
+        unique_key=["id", "ts"],
+    )
+    adapter.cursor.execute.assert_called_once_with(
+        'MERGE INTO "target" AS "__MERGE_TARGET__" USING (SELECT CAST("id" AS INTEGER) AS "id", CAST("ts" AS DATETIME2) AS "ts", CAST("val" AS INTEGER) AS "val" FROM (VALUES (1, 4), (2, 5), (3, 6)) AS "t"("id", "ts", "val")) AS "__MERGE_SOURCE__" ON "__MERGE_TARGET__"."id" = "__MERGE_SOURCE__"."id" AND "__MERGE_TARGET__"."ts" = "__MERGE_SOURCE__"."ts" '
+        'WHEN MATCHED THEN UPDATE SET "__MERGE_TARGET__"."id" = "__MERGE_SOURCE__"."id", "__MERGE_TARGET__"."ts" = "__MERGE_SOURCE__"."ts", "__MERGE_TARGET__"."val" = "__MERGE_SOURCE__"."val" '
+        'WHEN NOT MATCHED THEN INSERT ("id", "ts", "val") VALUES ("__MERGE_SOURCE__"."id", "__MERGE_SOURCE__"."ts", "__MERGE_SOURCE__"."val")'
+    )
+
+
+def test_replace_query_pandas(make_mocked_engine_adapter: t.Callable):
+    adapter = make_mocked_engine_adapter(MsSqlEngineAdapter)
+
+    df = pd.DataFrame({"a": [1, 2, 3], "b": [4, 5, 6]})
+    adapter.replace_query("test_table", df, {"a": "int", "b": "int"})
+
+    adapter.cursor.execute.assert_has_calls(
+        [
+            call('DROP TABLE IF EXISTS "test_table"'),
+            call('CREATE TABLE IF NOT EXISTS "test_table" ("a" int, "b" int)'),
+            call(
+                'INSERT INTO "test_table" ("a", "b") SELECT CAST("a" AS INTEGER) AS "a", CAST("b" AS INTEGER) AS "b" FROM (VALUES (1, 4), (2, 5), (3, 6)) AS "t"("a", "b")'
+            ),
+        ]
+    )
+
+
+def test_create_table_primary_key(make_mocked_engine_adapter: t.Callable):
```

Comment:
Can we also add a test for `create_index`. As far as I know MSSQL doesn't support `CREATE INDEX IF NOT EXISTS` statement I wonder whether sqlglot handles this correctly during transpilation.

Reply:
fwiw, it looks like this test succeeds but I do think sqlglot might not be parsing this correctly

Reply:
This is going to continue to generate a statement incompatible with MSSQL until a newer version of sqlglot can be used in the project. Aside from that, I think the code is in a pretty good spot for review.